R/tt_dotabulation.R

```diff
@@ -759,6 +759,18 @@ setMethod(".make_split_kids", "Split",
                  alt_dfpart$message,
                  call. = FALSE)
         }
+        
+        # Error if split does not have the same values in the alt_df (and order)
+        alt_df_spl_vals <- levels(as.factor(alt_df[[spl_payload(spl)]]))
+        # The following breaks if there are different levels (do_split returns empty list)
+        # or if there are different number of the same levels
+        if (length(alt_dfpart) != length(dataspl)) {
```

Comment:
I think we can (and should) make the requirement stronger. We don't just want the same number of facets, we want _**the same facets**_. Check if the sets of values are actually the same, rather than just having the same count.

Reply:
If the data does not contain the split values contained in the split (`spl`) which are defined in `df`, then the `do_split` on `alt_df` produces an empty list, giving a different length of obj. I can try to see what happens if there is a mix...

Reply:
covered!